Install.lua

```diff
@@ -1,175 +1,275 @@
+
+	if Website == "https://raw.githubusercontent.com/" then
+		if Directory then
+			Tree, Directory = Directory:match("^([^/]+)/(.+)")
+			if Directory then
+				ScriptName, ScriptClass = Directory:match("([%w-_%%]+)%.?(%l*)%.lua$")
+			end
+		end		
+	elseif Directory then
+		local a, b = Directory:find("^[tb][rl][eo][eb]/[^/]+")
+		if a and b then
+			Tree, Directory = Directory:sub(6, b), Directory:sub(b + 1)
+			if Directory == "" then Directory = nil end
+			if Directory and Link:find("blob") then
+				ScriptName, ScriptClass = Directory:match("([%w-_%%]+)%.?(%l*)%.lua$")
+			end
+		else
+			Directory = nil
+		end
+	end
+
+	if not Website then Website = "https://github.com/" end
+	Directory = Directory and ("/" .. Directory):gsub("^//", "/") or ""
+
+	-- Threads
+	local Routines = RoutineList or {false}
+	local Value = #Routines + 1
+	Routines[Value] = false
+
+	local Garbage
+
+	if ScriptName then
+		local Script = Instance.new(ScriptTypes[ScriptClass or "mod"])
+		if not Routines[1] then Routines[1] = Script end
+		Script.Name = ScriptName:gsub("Library$", "", 1):gsub("%%(%x%x)", UrlDecode)
+		local Link = "https://raw.githubusercontent.com/" .. Organization .. "/" .. Repository .. "/" .. (Tree or "master") .. Directory
+		DataSources[Script] = Link
+		Script.Source = GetAsync(Link)
+		Script.Parent = Parent and not RoutineList and Repository ~= ScriptName and GetFirstChild(Parent, Repository, "Folder") or Parent
+	elseif Repository then
+		Link = Website .. Organization .. "/" .. Repository .. ((Tree or Directory ~= "") and (Link .. "/tree/" .. (Tree or "master") .. Directory) or "")
+		if not Parent then Parent, Garbage = Instance.new("Folder"), true end
+		coroutine.resume(coroutine.create(InstallRepo), Link, 1, Parent, Routines) -- "/" .. Repository .. Directory
+	elseif Organization then
+		local Object = Instance.new("Folder")
+		Object.Name = Organization
+		Link = Website .. Organization
+		if not Routines[1] then Routines[1] = Object end
+		
+		for Link, Data in GetAsync(Link .. "?tab=repositories"):gmatch('<a href="(/' .. Organization .. '/[^/]+)" itemprop="name codeRepository">(.-)</div>') do
+			if not Data:match("Forked from") and not (Organization == "RoStrap" and Link:find("Plugin")) then
+				GitHub:Install(Link, Object, Routines)
+			end
+		end
+		Object.Parent = Parent
+	end
+
+	Routines[Value] = true
+
+	if not RoutineList then
+		repeat
+			local Done = 0
+			local Count = #Routines
+			for a = 1, Count do
+				if Routines[a] then
+					Done = Done + 1
+				end
+			end
+		until Done == Count or not wait()
+		local Object = Routines[1]
+		if Garbage then
+			Object.Parent = nil
+			Parent:Destroy()
+		end
+		DataSources[Object] = Link
+		return Object
+	end
 end
 
-HttpService.HttpEnabled = WasHttpEnabled
-ProgressText.Text = "Done!"
-done = true
+GitHub:Install("/Quenty/NevermoreEngine", game:GetService("ServerScriptService")).Name = "Nevermore"
+HttpService.HttpEnabled = ...
```

Comment:
Should leave a comment why we're doing this/usecase, incase someone tries to install this the other way.